### PR TITLE
Added averaging-methods guide

### PR DESCRIPTION
**Pull request by y27choi**
### Linked issue(s):
[KOL-2530](https://linear.app/kolena/issue/KOL-2530/port-averaging-method-doc-to-docskolenaio)

### What change does this PR introduce and why?
* Added Metrics Guide section in the main header
* Put up a placeholder for Metrics Guide page
* Added "Averaging Methods" to Metrics Guide section

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: docs/metrics-guide/index.md [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
-search:
-  exclude: true
 ---
 
 # :kolena-learning-20: Metrics Guide
+
+This section contains guides for different metrics. Metrics are used to measure the performance of a model. Each use case requires different metrics and finding the right metrics for your use case is very important to understand and compare the performance of your models. In each metrics guide, you can learn about the metric with examples, its limitations and biases, and its intended uses.
+
+<div class="grid cards" markdown>
+- [:octicons-container-24: Averaging Methods](averaging-methods.md)
```
Comment (y27choi) [inline docs/metrics-guide/index.md]: Probably not using the right icon here.  I wasn't sure where to pick one from.

**Comment by y27choi** [comment]
image
I couldn't tell why it has the "warning" sign looking icon next to Metrics Guide. Any idea?

File: docs/metrics-guide/averaging-methods.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Averaging Methods
```
Comment (gordonhart) [inline docs/metrics-guide/averaging-methods.md]: Can we include "macro" and "micro" in the title? It wasn't clear what this was to me at first glance. Maybe:

```
Macro and Micro Averaging
```

File: docs/metrics-guide/averaging-methods.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+| Classes | TP | FP | FN | Precision | Recall | F1-score |
+| --- | --- | --- | --- | --- | --- | --- |
+| Airplane | 2 | 1 | 1 | 0.67 | 0.67 | 0.67 |
+| Boat | 1 | 3 | 0 | 0.25 | 1.0 | 0.4 |
+| Car | 3 | 0 | 3 | 1.0 | 0.5 | 0.67 |
+
+**Macro-averaging** is perhaps the most straightforward among the numerous options and is computed by taking an **unweighted** mean of all the per-class scores:
+
+```python
+macro-avg F1 = (0.67 + 0.4 + 0.67) / 3 = 0.58
```
Comment (gordonhart) [inline docs/metrics-guide/averaging-methods.md]: Feels like these should use LaTeX for readability

File: docs/metrics-guide/averaging-methods.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+In contrast, **micro-averaging** computes a **global** average by counting the sums of true positive (TP), false negative (FN) and false positive (FP):
+
+```python
+micro-avg F1 = 6 / (6 + 0.5 * (4 + 4)) = 0.6
+```
+
+But what about `micro-avg precision` and `micro-avg recall`?
```
Comment (gordonhart) [inline docs/metrics-guide/averaging-methods.md]: Would be nice to have sections (maybe `###` sub-headers) for each of these averaging types, such that they show up in the sidebar and it's clear what this page is about.

```
### Macro Average

...

### Micro Average

...

### Weighted Average

...
```

File: docs/metrics-guide/averaging-methods.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+$$micro \space F1 = micro \space precision = micro \space recall = accuracy$$
+
+### Weighted Average
+**Weighted average** computes the mean of all per-class scores while considering each class’s **support**. In this case, support is the number of actual instances of the class in the dataset, for example, if there are 5 samples of class Airplane, then the support value of class Airplane is 5. In other words, support is the sum of TP and FN counts. The weight is the proportion of each class’s support relative to the sum of all support values:
+
+$$\begin{aligned}weighted \space F1 &= (0.67 \times 0.3) + (0.4 \times 0.1) + (0.67 \times 0.6) \\&= 0.64\end{aligned}$$
+
+### Samples Average
+**Samples average** computes the mean of metrics computed per sample. It’s only applicable to multi-label classification cases where a sample has multiple labels.
```
Comment (gordonhart) [inline docs/metrics-guide/averaging-methods.md]: Section is missing a formula?

**Comment by gordonhart** [comment]
> I couldn't tell why it has the "warning" sign looking icon next to Metrics Guide. Any idea?

This is the "new" status added at the top of `metrics-guide/index.md`. Feel free to remove.

**Review by gordonhart — APPROVED** [review]

**Pull request merged**